zuul-core/src/main/java/com/netflix/zuul/netty/connectionpool/DefaultClientChannelManager.java

```diff
@@ -341,12 +341,68 @@ public boolean remove(PooledConnection conn) {
             return promise;
         }
 
-        final InstanceInfo instanceInfo = chosenServer instanceof DiscoveryEnabledServer ?
-                ((DiscoveryEnabledServer) chosenServer).getInstanceInfo() :
-                // create mock instance info for non-discovery instances
-                new InstanceInfo(chosenServer.getId(), null, null, chosenServer.getHost(), chosenServer.getId(),
-                        null, null, null, null, null, null, null, null, 0, null, null, null, null, null, null, null, null, null, null, null, null);
+        String rawHost;
+        int port;
+        InstanceInfo instanceInfo;
+        if (chosenServer instanceof DiscoveryEnabledServer) {
+            DiscoveryEnabledServer discoveryServer = (DiscoveryEnabledServer) chosenServer;
+            // Configuration for whether to use IP address or host has already been applied in the
+            // DiscoveryEnabledServer constructor.
+            rawHost = discoveryServer.getHost();
+            port = discoveryServer.getPort();
+            instanceInfo = discoveryServer.getInstanceInfo();
+            // TODO(carl-mastrangelo): pull the IPv6 addr from the instance info, if present.
+        } else {
+            // create mock instance info for non-discovery instances
+            rawHost = chosenServer.getHost();
+            port = chosenServer.getPort();
+            instanceInfo = new InstanceInfo(
+                    chosenServer.getId(),
+                    null,
+                    null,
+                    chosenServer.getHost(),
+                    chosenServer.getId(),
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    0,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null);
+        }
 
+        InetSocketAddress serverAddr;
+        try {
+            InetAddress ipAddr = InetAddresses.forString(rawHost);
+            serverAddr = new InetSocketAddress(ipAddr, port);
+        } catch (IllegalArgumentException e1) {
+            LOG.warn("NettyClientConnectionFactory got an unresolved address, addr: {}", rawHost);
+            Counter unresolvedDiscoveryHost = SpectatorUtils.newCounter(
+                    "unresolvedDiscoveryHost",
+                    connPoolConfig.getOriginName() == null ? "unknownOrigin" : connPoolConfig.getOriginName());
+            unresolvedDiscoveryHost.increment();
+            try {
+                serverAddr = new InetSocketAddress(rawHost, port);
```

Comment:
Is this second round required because sometimes we may not get an IP above?

Reply:
Correct.   Zuul at Netflix doesn't run into this, but other code users might.   Just preserving any pre-existing behavior.